src/containers/guest-home-page/signup-dialog/SignupDialog.jsx

```diff
@@ -84,7 +85,24 @@ const SignupDialog = ({ type }) => {
   useEffect(() => {
     setNeedConfirmation(isDirty)
   }, [isDirty, setNeedConfirmation])
+  const breakpoints = useBreakpointsGoogle()
 
+  const { isSmallScreen, isMediumScreen, isLargeScreen, isXLargeScreen } =
+    useMemo(() => breakpoints, [breakpoints])
+
+  const buttonWidths = {
+    small: 330,
+    medium: 333,
+    large: 340,
+    xLarge: 394
+  }
+
+  const getButtonWidth = () => {
+    if (isSmallScreen) return buttonWidths.small
+    if (isMediumScreen) return buttonWidths.medium
```

Comment:
maybe it is better to make this function accept a breakpoints object and restructure it inside:

const breakpoints = useBreakpoints()

then, pass breakpoints to this function and restructure it there, also buttonWidth seems redundant, just return the width immidiately

Reply:
![image](https://github.com/ita-social-projects/SpaceToStudy-Client/assets/101183408/d284487b-5c08-4487-b3f8-3a9008fd0994)
this implementation looks better ?

Reply:
yeah, but move useBreakpoints to the top as hooks should be called in the top of a component

Reply:
![image](https://github.com/ita-social-projects/SpaceToStudy-Client/assets/101183408/505a2a93-0600-4dc1-b95b-ccb1d4d7cd19)
fixed